cmd/kubeadm/app/phases/downloadcerts/downloadcerts.go

```diff
@@ -0,0 +1,124 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package downloadcerts
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	certutil "k8s.io/client-go/util/cert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	certsutil "k8s.io/kubernetes/cmd/kubeadm/app/util/certs"
+	cryptoutil "k8s.io/kubernetes/cmd/kubeadm/app/util/crypto"
+)
+
+// FIXME (ereslibre): refactor with init
+const (
+	externalEtcdCA   = "external-etcd-ca.crt"
+	externalEtcdCert = "external-etcd.crt"
+	externalEtcdKey  = "external-etcd.key"
+)
+
+// DownloadCerts downloads the certificates needed to join a new control plane.
+func DownloadCerts(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, key []byte) error {
+	fmt.Printf("[download-certs] downloading the certificates in Secret %q in the %q Namespace\n", kubeadmconstants.KubeadmCertsSecret, metav1.NamespaceSystem)
+
+	secret, err := getSecret(client)
+	if err != nil {
+		return err
+	}
+
+	secretData, err := getSecretData(secret, key)
+	if err != nil {
+		return err
+	}
+
+	for certOrKeyName, certOrKeyPath := range certsToDownload(cfg) {
+		certOrKeyData, ok := secretData[certsutil.CertOrKeyNameToSecretName(certOrKeyName)]
+		if !ok {
+			return errors.New("couldn't find required certificate or key in Secret")
+		}
+		if err := writeCertOrKey(certOrKeyPath, certOrKeyData); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func writeCertOrKey(certOrKeyPath string, certOrKeyData []byte) error {
+	if _, err := certutil.ParsePrivateKeyPEM(certOrKeyData); err == nil {
+		return certutil.WriteKey(certOrKeyPath, certOrKeyData)
+	} else if _, err := certutil.ParseCertsPEM(certOrKeyData); err == nil {
+		return certutil.WriteCert(certOrKeyPath, certOrKeyData)
+	}
+	return errors.New("unknown data found in Secret entry")
+}
+
+func getSecret(client clientset.Interface) (*v1.Secret, error) {
+	secret, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(kubeadmconstants.KubeadmCertsSecret, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, errors.Errorf("Secret %q was not found in the %q Namespace. This Secret might have expired. Please, run `kubeadm init phase upload-certs` on a control plane to generate a new one", kubeadmconstants.KubeadmCertsSecret, metav1.NamespaceSystem)
+		}
+		return nil, err
+	}
+	return secret, nil
+}
+
+// FIXME (ereslibre): refactor with init: move to CertsToTransfer to app/util/certs package
+func certsToDownload(cfg *kubeadmapi.InitConfiguration) map[string]string {
```

Comment:
This should be the same function used for upload

Reply:
Yes, it's on the `FIXME` comment :)

Reply:
We can keep it on `certscopy` package.